components/Modal/CreateImageUpload.js

```diff
@@ -0,0 +1,751 @@
+                    id={key}
+                    key={key}
+                    onChange={this.setUploadSettings}
+                  />
+                </FormGroup>
+              ))}
+            </Form>
+          )}
+        </React.Fragment>
+      )
+    };
+
+    const uploadAzureStep = {
+      name: "Upload to Azure",
+      steps: [uploadAzureAuth, uploadAzureSettings]
+    };
+
+    const uploadVSphereAuth = {
+      name: "Authentication",
+      component: (
+        <React.Fragment>
+          {providerSettings.vsphere !== undefined && (
+            <Form isHorizontal>
+              {Object.keys(providerSettings.vsphere.auth).map(key => (
+                <FormGroup
+                  label={formatMessage({ id: "vsphere-auth", defaultMessage: providerSettings.vsphere.auth[key] })}
+                  fieldId={key}
+                  key={key}
+                >
+                  <TextInput
+                    value={this.state.uploadSettings[key]}
+                    type="text"
+                    id={key}
+                    key={key}
+                    onChange={this.setUploadSettings}
+                  />
+                </FormGroup>
+              ))}
+            </Form>
+          )}
+        </React.Fragment>
+      )
+    };
+
+    const uploadVSphereSettings = {
+      name: "File Upload",
+      component: (
+        <React.Fragment>
+          {providerSettings.vsphere !== undefined && (
+            <Form isHorizontal>
+              {Object.keys(providerSettings.vsphere.settings).map(key => (
+                <FormGroup
+                  label={formatMessage({
+                    id: "vsphere-settings",
+                    defaultMessage: providerSettings.vsphere.settings[key]
+                  })}
+                  fieldId={key}
+                  key={key}
+                >
+                  <TextInput
+                    value={this.state.uploadSettings[key]}
+                    type="text"
+                    id={key}
+                    key={key}
+                    onChange={this.setUploadSettings}
+                  />
+                </FormGroup>
+              ))}
+            </Form>
+          )}
+        </React.Fragment>
+      )
+    };
+
+    const uploadVSphereStep = {
+      name: "Upload to VSphere",
+      steps: [uploadVSphereAuth, uploadVSphereSettings]
+    };
+
+    const uploadOpenstackAuth = {
+      name: "Authentication",
+      component: (
+        <React.Fragment>
+          {providerSettings.openstack !== undefined && (
+            <Form isHorizontal>
+              {Object.keys(providerSettings.openstack.auth).map(key => (
+                <FormGroup
+                  label={formatMessage({
+                    id: "openstack-auth",
+                    defaultMessage: providerSettings.openstack.auth[key]
+                  })}
+                  fieldId={key}
+                  key={key}
+                >
+                  <TextInput
```

Comment:
This gives the following warning
A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component

I don't see what's wrong there in the first place, but maybe because you are assigning the state to another variable and using that, react gets confused thinking that the component in uncontrolled.

Reply:
This was due to the value being initialized to undefined and then later updated to a string. It is now fixed.